core/src/main/java/ai/timefold/solver/core/impl/domain/variable/provided/AbstractVariableReference.java

```diff
@@ -0,0 +1,23 @@
+package ai.timefold.solver.core.impl.domain.variable.provided;
+
+import ai.timefold.solver.core.preview.api.variable.provided.VariableReference;
+
+import org.jspecify.annotations.NonNull;
+import org.jspecify.annotations.Nullable;
+
+public abstract sealed class AbstractVariableReference<Solution_, Entity_, Value_> implements VariableReference<Entity_, Value_>
```

Comment:
It says class, but it doesn't actually have any concrete methods, which suggests it's an interface instead. :-) We typically call these interfaces `InnerX`, where `X` is the public thing that's being extended.